### DISCUSSION TITLE
AV1-HDR Question

### DISCUSSION THREAD

**Discussion by grndal** [cdgriffith/FastFlix#528] [General]
I'm a little confused.  I have a MKV file when played on my TV, pops up the HDR indicator.  When I run that file through FastFlix, the new file no longer pops up with the HDR indicator when played on my TV.  If I load the new file back into FastFlix, the video track is identified as a 10-bit bt2020 - HDR10.  My question is why is the TV no longer recognizing it as having HDR?  What option am I not selecting or what custom option would I need to add?  Any help would be appreciated.

Codec: AV1 (QSVEnvC)

Here is the raw command for the first 25 seconds.
`"C:\Videos\FastFlix Tools\QSVEncC\exe_files\QSVEncC\x64\QSVEncC64.exe" --avsw -i "X:\Orig.mkv" --video-metadata clear --metadata clear --video-metadata title="Orig" --chapter-copy -c av1 --cqp 28 --quality best --level auto --chromaloc auto --colorrange auto --colormatrix bt2020nc --transfer smpte2084 --colorprim bt2020 --master-display "G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(10000000,50)" --max-cll "1000,497" --dhdr10-info copy --output-depth 10 --avsync cfr --adapt-ref  --audio-copy 1,2,3,4,5,6 --audio-metadata 1?language=eng --audio-metadata 1?title="Surround 7.1" --audio-metadata 1?handler="Surround 7.1"  --audio-disposition 1?default --audio-metadata 2?language=eng --audio-metadata 2?title="Surround 5.1" --audio-metadata 2?handler="Surround 5.1"  --audio-disposition 2?unset --audio-metadata 3?language=eng --audio-metadata 3?title="Stereo" --audio-metadata 3?handler="Stereo"  --audio-disposition 3?unset --audio-metadata 4?language=eng --audio-metadata 4?title="Stereo" --audio-metadata 4?handler="Stereo"  --audio-disposition 4?unset --audio-metadata 5?language=eng --audio-metadata 5?title="Surround 5.1" --audio-metadata 5?handler="Surround 5.1"  --audio-disposition 5?unset --audio-metadata 6?language=eng --audio-metadata 6?title="Stereo" --audio-metadata 6?handler="Stereo"  --audio-disposition 6?unset  --sub-copy 1,2,3 --sub-disposition 1?unset --sub-metadata  1?language='eng' --sub-disposition 2?unset --sub-metadata  2?language='eng' --sub-disposition 3?unset --sub-metadata  3?language='eng' -m default_mode:infer_no_subs -o "C:\Videos\Converted\New.mkv"`

**Comment by cdgriffith** [comment]
Some TVs read that type of data differently (in container vs in HEVC/AV1 stream), and some just have issues with some containers. 

First would do a check with something like MediaInfo to make sure it also corroborates that it has the HDR10 flag.  If you set it to text mode, there should be a line like: 

```
HDR format                               : HDR10 compatible
```

If that looks good try adding it into a different container, like .mp4 or .ts. If that still doesn't work, there are tools (that I am honestly blanking on name of tonight) that can edit MKV containers and add HDR10 static metadata to the container itself. Would try that as last resort. 

I do not have a hardware device to test out any of the AV1 hardware encoders, so I honestly don't have any samples to help try and figure it out 😞 

Last thought, is the original file AV1 or HEVC? If it's HEVC the TV itself may just not understand AV1 with HDR as it's so new.

Let me know how that pans out!

**Comment by grndal** [comment]
Did some more digging and the issue is not related to FastFlix.  The MKV is being correctly generated with the HDR info and can be played directly on my TV without issue (including HDR).  What isn't working is when I play the file through Plex.  While Plex added AV1 support in 1.3, there appears to be an issue with the clients [not allowing direct streaming of AV1](https://forums.plex.tv/t/please-enable-av1-hardware-decoding-on-plex-client/823813/42) (Plex is transcoding to H264).  So, until Plex fixes their clients, AV1 will not work with direct streaming.

Thank you for the assist, but it looks like I'll need to wait on Plex or do some more converting on my side.  I've included my setup below.  If you ever need someone to test AV1 (files or settings) let me know.

**FastFlix/Plex Machine Hardware**
i7-13700K
32GB memory
Intel A770 Arc GPU
1TB SSD

**FastFlix/Plex Machine Software**
Windows 11
Plex 1.32.6.7557

**TV**
LG OLED77C2PUA
